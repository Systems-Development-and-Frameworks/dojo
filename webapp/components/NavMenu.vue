<template>
  <div class="flex" :class="vertical ? 'justify-start flex-col' : 'justify-between flex-row'">
    <NuxtLinkButton @click="$emit('navigated')" to="/" color="gray" href="/">Posts</NuxtLinkButton>
    <NuxtLinkButton @click="$emit('navigated')" to="/about" color="gray" href="/about">About us</NuxtLinkButton>
    <NuxtLinkButton @click="$emit('navigated')" v-if="!isAuthenticated" :to="{name: 'login', params: {returnPath: '/'}}"
      color="green" href="/login" id="login-button">
      Login
    </NuxtLinkButton>
    <BasicButton v-else :color="'red'" @click.native="$emit('logout')" id="logout-button">Logout</BasicButton>
  </div>
</template>
<script>
export default {
  name: "NavMenu",
  props: {
    isAuthenticated: {
      type: Boolean,
    },
    vertical: {
      type: Boolean,
      default: false
    }
  }
}
</script>

<style scoped>

</style>