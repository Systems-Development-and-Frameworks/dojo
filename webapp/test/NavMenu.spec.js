import '@testing-library/jest-dom'
import 'regenerator-runtime'

import { createLocalVue, shallowMount } from '@vue/test-utils'

import NavMenu from '@/components/NavMenu'

describe('NavMenu', () => {
    const localVue = createLocalVue()

    let wrapper

    const createLoginMenu = async (appOptions) => {
        wrapper = shallowMount(NavMenu, {
            localVue,
            stubs: {
                BasicButton: true,
                NuxtLinkButton: true,
            },
            ...appOptions
        })
        await wrapper.vm.$nextTick()
    }

    afterEach(() => {
        wrapper.destroy()
    })

    describe('for a user that is not logged in', () => {
        it('should show only the login button', async () => {
            await createLoginMenu()
            await wrapper.vm.$nextTick()

            expect(wrapper.find('#login-button').exists()).toBe(true)
            expect(wrapper.find('#logout-button').exists()).toBe(false)
            expect(wrapper.html()).toMatchSnapshot()
        })
    })

    describe('for a user that is logged in', () => {

        it('should show only the logout button and the user id', async () => {
            await createLoginMenu({ propsData: { isAuthenticated: true }})
            await wrapper.vm.$nextTick()

            expect(wrapper.find('#login-button').exists()).toBe(false)
            expect(wrapper.find('#logout-button').exists()).toBe(true)
            expect(wrapper.html()).toMatchSnapshot()
        })

    })
})

