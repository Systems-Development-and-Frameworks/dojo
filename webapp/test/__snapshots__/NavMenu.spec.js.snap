// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu for a user that is logged in should show only the logout button and the user id 1`] = `
"<div class=\\"flex justify-between flex-row\\">
  <nuxtlinkbutton-stub to=\\"/\\" color=\\"gray\\" href=\\"/\\">Posts</nuxtlinkbutton-stub>
  <nuxtlinkbutton-stub to=\\"/about\\" color=\\"gray\\" href=\\"/about\\">About us</nuxtlinkbutton-stub>
  <basicbutton-stub color=\\"red\\" id=\\"logout-button\\">Logout</basicbutton-stub>
</div>"
`;

exports[`NavMenu for a user that is not logged in should show only the login button 1`] = `
"<div class=\\"flex justify-between flex-row\\">
  <nuxtlinkbutton-stub to=\\"/\\" color=\\"gray\\" href=\\"/\\">Posts</nuxtlinkbutton-stub>
  <nuxtlinkbutton-stub to=\\"/about\\" color=\\"gray\\" href=\\"/about\\">About us</nuxtlinkbutton-stub>
  <nuxtlinkbutton-stub to=\\"[object Object]\\" color=\\"green\\" href=\\"/login\\" id=\\"login-button\\">
    Login
  </nuxtlinkbutton-stub>
</div>"
`;
